### PR TITLE
Add Entity objects to  common project

### PR DESCRIPTION
**Pull request by millerts**
I chose to move only the entity classes to minimize the impact of this change as the project is added to the build system and released.

These same classes will be used by Item Search Service and Item Ingest Service.  All classes were exactly the same as what is in Item Ingest Service develop branch.

### CODE REVIEW

File: build.gradle
Comment (millerts) [inline build.gradle:59]: I considered removing these test compile dependencies since the project at this time doesn't have tests.  I left them in because I figure there are more classes than just the entity classes that we'll share.

File: gradle.properties
Comment (millerts) [inline gradle.properties]: Should this be `0.1.0-SNAPSHOT`?

File: build.gradle
Comment (millerts) [inline build.gradle]: This line ensures gradle to check one's local repository before reaching out to the external repository.

File: build.gradle
Comment (millerts) [inline build.gradle:31]: This line ensures gradle to check one's local repository before reaching out to the external repository.

**Review by millerts — COMMENTED** [review]
This project is minimal.  Once approved I will merge and configure Bamboo to build the project/release.

File: build.gradle
Comment (swilson11) [inline build.gradle]: I this should be 1.5.9

File: build.gradle
Comment (swilson11) [reply build.gradle]: I get having it below, for pulling dependencies, but wouldn't we only need it up here if the build script itself was pulling dependencies (like gradle plugins) from local builds?

File: gradle.properties
Comment (swilson11) [reply gradle.properties]: 0.1.0 would be consistent with IIS and ISS

File: build.gradle
Comment (millerts) [reply build.gradle]: I defer to your judgement on how to proceed.  I was thinking that Gradle worked much like Maven which it will look at the local .m2 instead of pulling down the dependency each time a project is run.  I admit my lack of knowledge in this particular block so if that isn't advised I think we should remove it.

Can you please respond that you want me to remove it if it is there in error?  So I know what I should do.  I'll make the change in ingest as well if that is how we want to move forward.

File: gradle.properties
Comment (millerts) [reply gradle.properties]: Yeah, I noticed it after I pushed this PR and wanted to double check.  I'll make the change.  Thanks.

File: .gitignore
Comment (swilson11) [inline .gitignore]: I notice you don't exclude the idea "out" directory, do you configure idea to put it's build output somewhere else?

File: build.gradle
Comment (swilson11) [reply build.gradle]: I am about 95% sure it should be removed :-)

File: .gitignore
Comment (millerts) [reply .gitignore]: I didn't see it show up so maybe I'm not doing something right in my ide.  I think I built this project from command line more than within the ide.

I'll make the change.

File: build.gradle
Comment (millerts) [reply build.gradle]: Cool.  I'll remove it.  I'm still catching up on the Gradle from Maven move (on my part).

**Review by swilson11 — APPROVED** [review]

**Pull request merged**
